src/DIRAC/Core/Utilities/Proxy.py

```diff
@@ -225,20 +227,7 @@ def _putProxy(userDN=None, userName=None, userGroup=None, vomsFlag=None, proxyFi
   :returns: Tuple of originalUserProxy, useServerCertificate, executionLock
   """
   # Setup user proxy
-  if userDN:
-    userDNs = [userDN]
-  else:
-    result = getDNForUsername(userName)
-    if not result['OK']:
-      return result
-    userDNs = result['Value']  # a same user may have more than one DN
-
-  vomsAttr = ''
-  if vomsFlag:
-    vomsAttr = getVOMSAttributeForGroup(userGroup)
-
-  result = getProxy(userDNs, userGroup, vomsAttr, proxyFilePath)
-
+  result = getProxy(userDN or userName, userGroup, vomsFlag, proxyFilePath)
```

Comment:
This scenario is still possible, so I added on the `ProxyManager` service side: if the username is specified instead of the DN then:
```
userDNs = Registry.getDNsForUsernameInGroup(instance, group)['Value']
```
```
    # Use loop to fix a possible case of having several DNs for one user/group
    for userDN in userDNs:
      retVal = self.__proxyDB.getProxy(userDN, group, requiredLifeTime=requiredLifetime, voms=vomsAttribute)
      if retVal['OK']:
        break
      errors.append(retVal['Message'])

    if not retVal['OK']:
      return S_ERROR('; '.join(errors))
    chain, secsLeft = retVal['Value']
```